color/src/color.rs

```diff
@@ -939,4 +981,42 @@ mod tests {
             }
         }
     }
+
+    /// Test the claim in [`super::real_round_to_u8`] that the only rounding failure in the range
+    /// of interest occurs for `0.49999997`.
+    #[test]
+    fn fast_round() {
+        #[expect(clippy::cast_possible_truncation, reason = "deliberate quantization")]
+        fn real_round_to_u8(v: f32) -> u8 {
+            v.round() as u8
+        }
+
+        // Check the rounding behavior at integer and half integer values within (and near) the
+        // range 0-255, as well as one ULP up and down from those values.
+        let mut failures = alloc::vec![];
+        let mut v = -1_f32;
+
+        while v <= 256. {
+            // Note we don't get accumulation of rounding errors by incrementing with 0.5: integers
+            // and half integers are exactly representable in this range.
+            assert!(v.abs().fract() == 0. || v.abs().fract() == 0.5, "{v}");
+
+            let down = v.next_down();
+            let up = v.next_up();
+
+            if real_round_to_u8(down) != fast_round_to_u8(down) {
+                failures.push(down);
+            }
+            if real_round_to_u8(v) != fast_round_to_u8(v) {
+                failures.push(v);
+            }
+            if real_round_to_u8(up) != fast_round_to_u8(up) {
+                failures.push(up);
+            }
```

Comment:
Sorry to be picky, but I might prefer us to go two ULPs in each direction (i.e. to validate that `0.49999994` has the correct behaviour). I think that the main testing block can be made into a closure, so it would look a bit like:

```
let validate_rounding = |val: f32| if real_round_to_u8(val) != fast_round_to_u8(val) {failures.push(val)};

validate_rounding(v.next_down().next_down());
validate_rounding(v.next_down());
validate_rounding(v);
validate_rounding(v.next_up());
validate_rounding(v.next_up().next_up());
```